core/domain/summary_services_test.py

```diff
@@ -764,3 +766,50 @@ def test_for_recently_published_explorations(self):
             recently_published_exploration_summaries[1]['title'], 'New title')
         self.assertDictContainsSubset(
             test_summary_3, recently_published_exploration_summaries[0])
+
+
+class ActivityReferenceValidationTests(test_utils.GenericTestBase):
+    """Tests for validation of activity references."""
+
+    EXP_ID_0 = 'exp_id_0'
+    EXP_ID_1 = 'exp_id_1'
+    COL_ID_2 = 'col_id_2'
+
+    def setUp(self):
+        super(ActivityReferenceValidationTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+
+    def test_validation_of_activity_ids_list(self):
```

Comment:
Maybe:

- ``test_requiring_non_existent_activities_to_be_public_raises_exceptions``
- ``test_requiring_private_activities_to_be_public_raises_exceptions``
- ``test_requiring_public_activities_to_be_public_succeeds``

And split up the test accordingly.